core/services/vrf/delegate.go

```diff
@@ -98,9 +100,14 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.Service, error) {
 	lV1 := l.Named("VRFListener")
 	lV2 := l.Named("VRFListenerV2")
 
+	latestHead, err := chain.Client().HeadByNumber(context.Background(), nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
@samsondav I've had to resort to this approach because the `chain.HeadTracker().LatestChain()` is pretty much impossible to mock out at the moment with the architecture of the head tracker / saver the way it is right now.

Reply:
I think we can just grab the latest head from the db in the resp count query to avoid this. Worst case is you end up loading some older than finality resp counts, but thats ok, they'll just never be re-rolled